app/controllers/forumthreads_controller.rb

```diff
@@ -80,6 +85,20 @@ def destroy
     redirect_to @thread.forum
   end
 
+  def search
+  end
+
+  def search_redirect
```

Comment:
Not to mention every `if [title, content, reply, query].any?` and anything like it will have to be changed into something such as `if title != "" || content != ""`...etc. Or I could nil empty parameters at the start of the search to avoid all of that, which would overall increase speed. However, nilling parameters in the model would be the same as doing so in the controller, so I would just as well nil them in the controller and redirect to clear them out of the URL.